app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -307,102 +309,113 @@ public void resendConsentAgreement(Study study, SubpopulationGuid subpopGuid, St
     }
-    
+
     public List<NotificationRegistration> listRegistrations(Study study, String userId) {
         checkNotNull(study);
         checkNotNull(userId);
-        
+
         Account account = getAccountThrowingException(study, userId);
-        
+
         return notificationsService.listRegistrations(account.getHealthCode());
     }
-    
+
     public void sendNotification(Study study, String userId, NotificationMessage message) {
         checkNotNull(study);
         checkNotNull(userId);
         checkNotNull(message);
-        
+
         Account account = getAccountThrowingException(study, userId);
 
         notificationsService.sendNotificationToUser(study.getStudyIdentifier(), account.getHealthCode(), message);
     }
-    
+
     private IdentifierHolder saveParticipant(Study study, Set<Roles> callerRoles, StudyParticipant participant,
             boolean isNew, boolean sendVerifyEmail) {
         checkNotNull(study);
         checkNotNull(callerRoles);
         checkNotNull(participant);
-        
+
         Validate.entityThrowingException(new StudyParticipantValidator(study, isNew), participant);
+        
+        boolean newAccountAssigningId = isNew && study.isExternalIdValidationEnabled()
+                && isNotBlank(participant.getExternalId());
         Account account = null;
         if (isNew) {
             // Don't set it yet. Create the user first, and only assign it if that's successful.
             // Allows us to assure that credentials and ID will be related or not created at all.
-            if (isNotBlank(participant.getExternalId())) {
-                externalIdService.reserveExternalId(study, participant.getExternalId());    
+            if (newAccountAssigningId) {
+                externalIdService.reserveExternalId(study, participant.getExternalId());
             }
             account = accountDao.constructAccount(study, participant.getEmail(), participant.getPassword());
         } else {
             account = getAccountThrowingException(study, participant.getId());
-            
-            addValidatedExternalId(study, participant, account.getHealthCode());
-        }
-        Map<ParticipantOption,String> options = Maps.newHashMap();
-        for (ParticipantOption option : ParticipantOption.values()) {
-            options.put(option, option.fromParticipant(participant));
-        }
-        // If we're validating the ID, we do this through the externalIdService, which writes to the participant options
-        // table when its appropriate to do so
-        if (study.isExternalIdValidationEnabled()) {
-            options.remove(EXTERNAL_IDENTIFIER);
+
+            updateValidatedExternalId(study, participant, account.getHealthCode());
         }
+
+        // Save all participant options except for the external ID, if we are going to use the registration service
+        Map<ParticipantOption, String> options = getAllOptionsButExternalId(study, participant);
```

Comment:
This comment used to say

        // If we're validating the ID, we do this through the externalIdService, which writes to the participant options
        // table when its appropriate to do so

The new comment is much more confusing. Case in point, I spent about 15 minutes tracing through the new code, old code, new tests, old tests, until I found this comment and figured out what was going on.